crates/forge/src/test_execution_syscall_handler.rs

```diff
@@ -691,18 +679,11 @@ fn execute_syscall(
     constants: &HashMap<String, Felt252>,
     cheatable_syscall_handler: &mut CheatableSyscallHandler,
 ) -> Result<(), HintError> {
-    let (cell, offset) = extract_buffer(system);
-    let system_ptr = get_ptr(vm, cell, &offset)?;
-
-    cheatable_syscall_handler
-        .child
-        .verify_syscall_ptr(system_ptr)?;
-
-    // We peek into memory to check the selector
-    let selector = DeprecatedSyscallSelector::try_from(felt_to_stark_felt(
-        &vm.get_integer(cheatable_syscall_handler.child.syscall_ptr)
-            .unwrap(),
-    ))?;
+    let selector = DeprecatedSyscallSelector::try_from(get_syscall_selector(
```

Comment:
sidenote: Why is this one `DeprecatedSyscallSelector` 🤔

Reply:
I think we should use SyscallSeclector here instead